testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -76,7 +71,7 @@ impl PerformanceBenchmarkParams {
 impl ExperimentParam for PerformanceBenchmarkParams {
     type E = PerformanceBenchmark;
     fn build(self, cluster: &Cluster) -> Self::E {
-        if self.is_fullnode {
+        if cluster.fullnode_instances().len() >= cluster.validator_instances().len() {
```

Comment:
We should probably replace `Cluster` with some `struct ExeprimentInitContext{cluster, global_emit_job_request} in `experiments::get_experiment->..->ExperimentParams::build`

Reply:
Since we're already passing it in the `run` function, I think it makes sense to use it there. The split of up/down instances can be decided at run time rather than build time.